### PR TITLE
fix: update tooling book

### LINKED ISSUE #14: Fix `#> Error in key(x$DT): could not find function "key"` in archive chapter
Full message:
```
#> Error in key(x$DT): could not find function "key"
```

The [vignette it was based on](https://cmu-delphi.github.io/epiprocess/articles/archive.html) attached `data.table` which made this line work, but the chapter must not.

### LINKED ISSUE #16: Incorporate syntax updates
The name/structure of some functions in the epi libraries seem to have changed, so parts of the tooling book's code do not currently work.

 In general:

- [x] Change `covidcast()` to `pub_covidcast()`, and adjust arguments.
- [x] Change `pivot_quantiles` to `pivot_quantiles_wider`.
- [x] Change `levels` to `quantile_levels` in `arx_args_list`.
- [x] Change `probs` to `quantile_levels` in `layer_residual_quantiles`, and `flatline_args_list`.
- [x] Change `tau` to `quantile_levels` in `quantile_reg`.

Chapter 2:

- [x] At [line](https://github.com/cmu-delphi/delphi-tooling-book/blob/4c3830c60a427abf010f890f315748f542e7584a/epidf.qmd#L52) get
#Warning messages:
1: epi_archive constructor argument `geo_type` is now ignored. Consider removing. 
2: epi_archive constructor argument `time_type` is now ignored. Consider removing. 

Chapter 6:

- [x] [Error](https://github.com/cmu-delphi/delphi-tooling-book/blob/4c3830c60a427abf010f890f315748f542e7584a/outliers.qmd#L70) because NULL is not accepted anymore for `seasonal_period`
- [x] [Error](https://github.com/cmu-delphi/delphi-tooling-book/blob/4c3830c60a427abf010f890f315748f542e7584a/outliers.qmd#L91) with `unnest(outlier_info)`

Chapter 7:

- [x] [Error](https://github.com/cmu-delphi/delphi-tooling-book/blob/4c3830c60a427abf010f890f315748f542e7584a/archive.qmd#L50) because `archive_cases_dv_subset_dt` does not exist

### PR DESCRIPTION
**Pull request by dshemetov**
Updating for changes here: https://github.com/cmu-delphi/epiprocess/pull/477.

Fixes #16.
Fixes #14.

### CODE REVIEW

File: _freeze/sliding-forecasters/figure-html/plot-ar-asof-1.svg
Comment (dajmcdon) [inline _freeze/sliding-forecasters/figure-html/plot-ar-asof-1.svg:1]: Something is wrong with the lower (blue) plot. It looks like multiple forecasts per target date connected.

**Review by dajmcdon — APPROVED** [review]
One figure should probably be fixed, but the rest looks good to me. Thanks for doing the updates.

File: _freeze/sliding-forecasters/figure-html/plot-ar-asof-1.svg
Comment (dshemetov) [reply _freeze/sliding-forecasters/figure-html/plot-ar-asof-1.svg:1]: I had the same thought and I think I made sure it didn't have multiple target dates, but I'll double check

Comment (dshemetov) [reply _freeze/sliding-forecasters/figure-html/plot-ar-asof-1.svg:1]: Fixed!

**Pull request merged**